### PR TITLE
Refactor contact stuff

### PR DESCRIPTION
**Pull request by moxie0**
1) Create a TextSecure 'account' with a mime type visible from a system contact card.
2) Fix all the speed issues associated with querying contacts.
3) Don't create a new canonical address id for *every single character* typed into the new conversation activity search field.
4) Restyle the fragment used by NewConversationActivity and PushContactSelectionActivity.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ContactSelectionActivity.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        } catch (IOException e) {
+          Log.w(TAG, e);
+        }
+
+        return null;
+      }
+
+      @Override
+      protected void onPostExecute(Void result) {
+        searchText.setText("");
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ContactSelectionActivity.java]: if DirectoryHelper.refreshDirectory is as long an operation as I think it is, we might want to make sure the activity is still around before modifying UI elements and telling fragments to do stuff.

File: src/org/thoughtcrime/securesms/ContactSelectionListFragment.java [hunk 2]
```diff
@@ -82,7 +80,17 @@ public void onPause() {
 
   @Override
   public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-    return inflater.inflate(R.layout.push_contact_selection_list_activity, container, false);
+    View view = inflater.inflate(R.layout.contact_selection_list_fragment, container, false);
+
+    this.emptyText = (TextView)                  view.findViewById(android.R.id.empty);
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ContactSelectionListFragment.java]: to this or not to this

File: src/org/thoughtcrime/securesms/ContactSelectionActivity.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    }.execute();
+  }
+
+  @Override
+  public void onContactSelected(String number) {}
+
+  private static class SearchUtil {
+
+    public static boolean isTextInput(EditText editText) {
+      return (editText.getInputType() & 0x0000000F) == InputType.TYPE_CLASS_TEXT;
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ContactSelectionActivity.java]: InputType.TYPE_MASK_CLASS

File: src/org/thoughtcrime/securesms/ContactSelectionActivity.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    public static boolean isPhoneInput(EditText editText) {
+      return (editText.getInputType() & 0x0000000F) == InputType.TYPE_CLASS_PHONE;
+    }
+
+    public static boolean isEmpty(EditText editText) {
+      return editText.getText().length() <= 0;
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ContactSelectionActivity.java]: can getText() ever be null?

**Comment by agrajaghh** [comment]
maybe dumb question: wasn't your plan to add all the new permissions at once to not annoy the users multiple times?

**Comment by moxie0** [comment]
in 2.23.0

**Comment by moxie0** [comment]
@agrajaghh we will

**Pull request closed without merging**